tests/integration_tests/cloud/test_cloud_workspaces.py

```diff
@@ -15,16 +18,57 @@ def test_deploy_source(
     cloud_workspace: CloudWorkspace,
 ) -> None:
     """Test deploying a source to a workspace."""
-    source = ab.get_source(
+    local_source: ab.Source = ab.get_source(
         "source-faker",
         local_executable="source-faker",
-        config={"count": 100},
+        config={"count": 100, "seed": 123},
         install_if_missing=False,
     )
-    source.check()
-    source_id: str = cloud_workspace._deploy_source(source)
+    local_source.check()
+
+    # Deploy source:
+    source_connector: cloud.CloudConnector = cloud_workspace.deploy_source(
+        local_source,
+        name_key="My Faker Source",  # Used in deduplication and idempotency
+        update_existing=False,       # Fail if source already exists
+    )
+    assert source_connector.source_name == "My Faker Source"
+    assert source_connector.configuration["count"] == 100
+
+    with raises(PyAirbyteResourceConflictError):
+        # Deploy source again (should fail):
+        cloud_workspace.deploy_source(
+            source=local_source,
+            name_key="My Faker Source",  # Used in deduplication and idempotency
+            update_existing=False,       # Fail since source already exists
+        )
+
+    # Change config and deploy source again (should succeed):
+    local_source.set_config({"count": 200})
+    source_connector: cloud.CloudConnector = cloud_workspace.deploy_source(
+        source=local_source,
+        name_key="My Faker Source",  # Used in deduplication and idempotency
+        update_existing=True,        # Update existing source
+    )
+
+    # Partially update the configuration:
+    source_connector.update_configuration(
+        {"count": 300},
+        merge=True,
```

Comment:
@aaronsteers  How do you envision `merge` being different from `update_existing`? 
In my mind, those sound like the same thing. Can we use `update_existing` in both places?

Reply:
Hi, @alex-gron. The "update_existing" was/is planned for a "deploy*()" verb - where a `False` value would fail if the resource already exists - rather than update it.

In this context "merge" is in the context of changes to settings - where 'merge=true' means we want to merge the provided setting with existing settign, and 'merge=false' would mean we replace all non-provided keys with the provided ones.